src/ui/public/vis/agg_configs.js

```diff
@@ -145,14 +144,45 @@ export default function AggConfigsFactory(Private) {
     });
 
     removeParentAggs(dslTopLvl);
-
     return dslTopLvl;
   };
 
   AggConfigs.prototype.getRequestAggs = function () {
-    return _.sortBy(this, function (agg) {
+
+
+    const aggregations = _.sortBy(this, function (agg) {
       return agg.schema.group === 'metrics' ? 1 : 0;
     });
+
+
+    let shouldAddGeocentroid = false;
```

Comment:
It doesn't feel like this is where we should be adding this, prior to this change we don't have anything hardcoded for a specific aggregation type here.